src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -922,4 +923,58 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise. A date is valid if there is no dueDateError or assessmentDueDateError
+     * @throws BadRequestException if the dates are not valid
+     */
+    public void validateDates() throws BadRequestException {
+        // All fields are optional, so there is no error if none of them is set
+        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null) {
+            return;
+        }
+        // at least one is set, so we have to check the two possible errors
+        boolean validDates = validateDueDate(getReleaseDate(), getDueDate()) && validateAssessmentDueDate(getReleaseDate(), getDueDate(), getAssessmentDueDate());
+
+        if (!validDates) {
+            throw new BadRequestException();
+        }
+    }
+
+    /**
+     * This method is used to validate the dueDate of an exercise. A due date is valid, if it is after the releaseDate,
+     * @return true if there is no DueDateError
+     */
+    private boolean validateDueDate(ZonedDateTime releaseDate, ZonedDateTime dueDate) {
+        if (releaseDate == null) {
+            return true;
+        }
+        return isBeforeAndNotNull(releaseDate, dueDate);
+    }
+
+    /**
+     * This method is used to validate the assesmentDueDate of an exercise. An assessmentDueDate is valid if it is after the releaseDate and dueDate. A given assesmentDueDate is invalid without an according dueDate
+     * @return true if there is no assessmentDueDateError
+     */
+    private boolean validateAssessmentDueDate(ZonedDateTime releaseDate, ZonedDateTime dueDate, ZonedDateTime assessmentDueDate) {
+        if (assessmentDueDate == null) {
+            return true;
+        }
```

Comment:
This part is also checked as part of `isBeforeAndNotNull` and is not necessarily needed. 
In this instance, however, having this explicit check might be useful as (a) the method can exit within one check and (b) the behavior is more explicit.